java-chassis-spring-boot/spring-boot2-starter-parent/spring-boot2-starter-gateway/src/main/java/org/apache/servicecomb/springboot2/starter/gateway/ScbDiscoveryClient.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.servicecomb.springboot2.starter.gateway;
+
+import org.apache.servicecomb.foundation.common.net.URIEndpointObject;
+import org.apache.servicecomb.springboot.common.AbstractDiscoveryClient;
+import org.apache.servicecomb.springboot.common.InstanceDiscoveryFilter;
+import org.springframework.cloud.client.DefaultServiceInstance;
+import org.springframework.cloud.client.ServiceInstance;
+import org.springframework.cloud.client.discovery.DiscoveryClient;
+
+import java.util.List;
+
+public class ScbDiscoveryClient extends AbstractDiscoveryClient implements DiscoveryClient {
```

Comment:
seems many files duplicated with project java-chassis-spring-boot/spring-boot2-starter-parent/spring-boot2-starter-discovery
maybe can depend on it directly?

Reply:
OK, spring-boot2-starter-gateway can depend on spring-boot2-starter-discovery directly.And spring-boot2-starter-gateway will only have a pom.xml file, but the gateway forward confuction is OK.

Reply:
you can use rebase to avoid too many useless commit log.